### PR TITLE
Tdex flow

### PR DESCRIPTION
**Pull request by tiero**
Ref: https://github.com/opendexnetwork/opendex.app/pull/18

Notable changes for y'all:

* The port for development is now 8080, not 3000 by default. I hope everyone is OK with this, since Nigiri box got that port already and was clashing with this.

* I introduced [tdex-react-scripts](https://www.npmjs.com/package/tdex-react-scripts) which is basically a fork of create react app, with the only add to webpack config to go read/compile Typescript also in the node_modules folder of `tdex-protobuf`. This allow us to not eject from react-scripts. Its literally one line change so pretty easy to verifify and keep in sync with upstream 

* Added try/catch in `ChooseTradingPair`'s  `On{Send/Receive}AmountChange` and add the error near the input form via `set{Send,Receive}AmountError` state

* make sure the Next button is disabled when `isPreviewing`

* Added npm scripts for provider related machinery, with the format `yarn provider:<boltz|comit|tdex>`  to run and `yarn clean:<>` to clean it

cc/ @opendexnetwork/opendex-app

### CODE REVIEW

**Comment by tiero** [comment]
First swap :) https://blockstream.info/liquid/tx/c3b57b46e435c56a8c0f4c00672e44195c1f76cc1ae21f692b82a022c994868d

File: package.json [hunk 1]
```diff
@@ -31,24 +34,26 @@
...
     "redux-observable": "^1.2.0",
     "redux-saga": "^1.1.3",
     "reselect": "^4.0.0",
     "rxjs": "^6.6.6",
+    "tdex-sdk": "^1.6.7",
     "web3modal": "^1.9.3"
   },
   "scripts": {
-    "start": "BROWSER=none react-scripts start",
+    "start": "BROWSER=none PORT=8080 react-scripts start",
```
Comment (da-kami) [inline package.json:46]: OK with me

File: package.json [hunk 2]
```diff
@@ -31,24 +34,26 @@
...
     "build": "react-scripts build",
     "test": "yarn prettier:check && react-scripts test",
     "eject": "react-scripts eject",
     "storybook": "start-storybook -p 6006 -s public",
     "storybook:no-cache": "start-storybook -p 6006 -s public --no-manager-cache",
     "build-storybook": "build-storybook -s public",
-    "prettier": "prettier --write src public",
-    "prettier:check": "prettier --check src public"
+    "prettier": "npx prettier --write src public",
+    "prettier:check": "npx prettier --check src public",
```
Comment (da-kami) [inline package.json:54]: Any particular reason why you use `npx` here?

File: src/components/ChooseTradingPair/index.tsx [hunk 3]
```diff
@@ -132,24 +135,25 @@ const ChooseTradingPair = (_props: ChooseTradingPairProps) => {
...
+          [sendCurrency.id, receiveCurrency.id]
+        );
+        dispatch(
+          setReceiveAmount(
+            convertAmountToString(receiveValue.amountWithFees.amount)
+          )
+        );
+        validateReceiveLimits(receiveValue.amountWithFees.amount);
+      } catch (e: any) {
+        setSendAmountError(e.message);
```
Comment (da-kami) [inline src/components/ChooseTradingPair/index.tsx:154]: I did something similar, see: https://github.com/opendexnetwork/opendex.app/pull/36/files#diff-79edee4c28d4b53cf2d649ac529fa993203be7d2340be547c9f880c9d102b406R187-R211 

But in order to get the state working properly I also had to reset the error to null if there was an error and then it works again. 
You did not run into that problem? 
Generally, I think we should eventually make the ChooseTradingPair component more data driven and at some point add a Reducer to cover the state changes.

**Review by da-kami — COMMENTED** [review]
Note: I did not give your flow a review, just briefly had a look at the points that concern the other swap providers and added some comments on the way. 
Overall I don't see any blockers for merging this!

File: package.json (hunk 2 above)
Comment (tiero) [reply package.json:54]: Want to make sure to use the installed prettier package in the project node modules, not the global one (just to have anyone to reproduce with same module)

File: src/components/ChooseTradingPair/index.tsx (hunk 3 above)
Comment (tiero) [reply src/components/ChooseTradingPair/index.tsx:154]: > add a Reducer to cover the state changes

 at the moment I just followed what we have down in master, so didn't changed much the  structure which should be done in a dedicated PR I would say,

**Pull request merged**